src/components/AutoUpdateTime.js

```diff
@@ -0,0 +1,99 @@
+import {View} from 'react-native';
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import styles from '../styles/styles';
+import DateUtils from '../libs/DateUtils';
+import withLocalize, {withLocalizePropTypes} from './withLocalize';
+import Timers from '../libs/Timers';
+import Text from './Text';
+
+const propTypes = {
+    /** Timezone of the user from their personal details */
+    timezone: PropTypes.shape({
+        /** Value of selected timezone */
+        selected: PropTypes.string,
+
+        /** Whether timezone is automatically set */
+        automatic: PropTypes.bool,
+    }).isRequired,
+    ...withLocalizePropTypes,
+};
+
+class AutoUpdateTime extends PureComponent {
+    constructor(props) {
+        super(props);
+        this.getCurrentUserLocalTime = this.getCurrentUserLocalTime.bind(this);
+        this.updateCurrentTime = this.updateCurrentTime.bind(this);
+        this.getTimezoneName = this.getTimezoneName.bind(this);
+        this.state = {
+            timezone: this.getCurrentUserLocalTime(),
+        };
+    }
+
+    componentDidMount() {
+        this.updateCurrentTime();
+    }
+
+    componentDidUpdate() {
+        // Make sure the interval is up to date everytime component updated.
```

Comment:
```suggestion
        // Make sure the interval is up to date every time the component updates
```
Leaving a suggestion here but Tthis comment should explain *why*. Sorry for not catching this earlier